File: backend/components/metiers/administration_publique/tests/samples/sample_utils.py
# sample_utils.py – Exemples ultra avancés d’utilitaires pour tests, audit, conformité, RGPD
from ..utils.core.core_utils import generate_id, audit_log
from ..utils.i18n.i18n_utils import translate
from ..utils.rbac.rbac_utils import has_permission

def sample_user():
    return {
        'id': generate_id('user'),
        'username': 'sampleuser',
        'email': 'sample@dihya.io',
        'roles': ['admin', 'auditor']
    }

def sample_audit_action():
    user = sample_user()
    return audit_log('SAMPLE_ACTION', user['id'], {'lang': translate('audit', 'fr')})

def sample_permission_check():
    user = sample_user()
    return has_permission(user, 'audit')

File: backend/components/metiers/administration_publique/tests/samples/sample_utils.js
// sample_utils.js – Exemples ultra avancés d’utilitaires pour tests, audit, conformité, RGPD
const { generateId, auditLog } = require('../utils/core/core_utils');
const { translate } = require('../utils/i18n/i18n_utils');
const { hasPermission } = require('../utils/rbac/rbac_utils');

function sampleUser() {
  return {
    id: generateId('user'),
    username: 'sampleuser',
    email: 'sample@dihya.io',
    roles: ['admin', 'auditor']
  };
}

function sampleAuditAction() {
  const user = sampleUser();
  return auditLog('SAMPLE_ACTION', user.id, { lang: translate('audit', 'fr') });
}

function samplePermissionCheck() {
  const user = sampleUser();
  return hasPermission(user, 'audit');
}

module.exports = {
  sampleUser,
  sampleAuditAction,
  samplePermissionCheck
};
